lumicks/pylake/channel.py

```diff
@@ -304,7 +304,28 @@ def downsampled_like(self, other_slice, reduce=np.mean):
             ]
         )
 
-        return self._with_data_source(TimeSeries(downsampled, timestamps))
+        new_slice = self._with_data_source(TimeSeries(downsampled, timestamps))
+
+        # the reconstructed slice may not cover the entire time range of
+        # the reference slice due to missing HF data. Crop the reference slice
+        # so that both slices have exactly the same timestamps
+        def crop_slice(full_slice, start_time, stop_time):
+            overlap_indices = np.argwhere(
+                np.logical_and(
+                    full_slice.timestamps >= start_time, full_slice.timestamps <= stop_time
+                )
+            ).squeeze()
+
+            overlap_data = full_slice.data[overlap_indices]
+            overlap_timestamps = full_slice.timestamps[overlap_indices]
+            return full_slice._with_data_source(TimeSeries(overlap_data, overlap_timestamps))
+
+        time_limits = (
+            np.max([new_slice.timestamps[0], other_slice.timestamps[0]]),
+            np.min([new_slice.timestamps[-1], other_slice.timestamps[-1]]),
+        )
+
+        return crop_slice(new_slice, *time_limits), crop_slice(other_slice, *time_limits)
```

Comment:
Why are you cropping the new slice? Isn't it by definition always smaller?

Reply:
Would it be easier to do something like this or am I overlooking something?

```python
        start_idx = np.searchsorted(timestamps - delta_time[0], self._src.start)
        stop_idx = np.searchsorted(timestamps, self._src.stop)
        timestamps_sliced = timestamps[start_idx:stop_idx]
        delta_time = delta_time[start_idx:stop_idx]

        downsampled = np.array(
            [
                reduce(self[start:stop].data)
                for start, stop in zip(timestamps_sliced - delta_time, timestamps_sliced)
            ]
        )

        new_slice = self._with_data_source(TimeSeries(downsampled, timestamps_sliced))

        return new_slice, other_slice[timestamps[start_idx]:timestamps[stop_idx]]
```

Reply:
ugh, I apparently forgot you could slice `Slice`s ... it's much simpler now